python/lsst/cp/verify/verifyStats.py

```diff
@@ -0,0 +1,510 @@
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with components:
+            - ``outputStats`` : `dict`
+                The output measured statistics.
+
+        Notes
+        -----
+        The outputStats should have a yaml representation of the form
+
+        AMP:
+          Amp1:
+            STAT: value
+            STAT2: value2
+          Amp2:
+          Amp3:
+        DET:
+          STAT: value
+          STAT2: value
+        CATALOG:
+          STAT: value
+          STAT2: value
+        VERIFY:
+          DET:
+            TEST: boolean
+          CATALOG:
+            TEST: boolean
+          AMP:
+            Amp1:
+              TEST: boolean
+              TEST2: boolean
+            Amp2:
+            Amp3:
+        SUCCESS: boolean
+
+        """
+        outputStats = {}
+
+        if self.config.doVignette:
+            VignetteExposure(inputExp, doUpdateMask=True, maskPlane='BAD',
+                             doSetValue=False, log=self.log)
+
+        mask = inputExp.getMask()
+        maskVal = mask.getPlaneBitMask(self.config.maskNameList)
+        statsControl = afwMath.StatisticsControl(self.config.numSigmaClip,
+                                                 self.config.clipMaxIter,
+                                                 maskVal)
+
+        # This is wrapped below to check for config lengths, as we can
+        # make a number of different image stats.
+        outputStats['AMP'] = self.imageStatistics(inputExp, statsControl)
+        if len(self.config.catalogStatKeywords):
+            outputStats['CATALOG'] = self.catalogStatistics(inputExp, statsControl)
+        else:
+            outputStats['CATALOG'] = {}
+        if len(self.config.detectorStatKeywords):
+            outputStats['DET'] = self.detectorStatistics(outputStats, statsControl)
+        else:
+            outputStats['DET'] = {}
+
+        outputStats['VERIFY'], outputStats['SUCCESS'] = self.verify(inputExp, outputStats)
+
+        return pipeBase.Struct(
+            outputStats=outputStats,
+        )
+
+    @staticmethod
+    def _emptyAmpDict(exposure):
+        """Construct empty dictionary indexed by amplifier names.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to extract detector from.
+
+        Returns
+        -------
+        outputStats : `dict` [`str`, `dict`]
+            A skeleton statistics dictionary.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if no detector can be found.
+        """
+        outputStats = {}
+        detector = exposure.getDetector()
+        if detector is None:
+            raise RuntimeError("No detector found in exposure!")
+
+        for amp in detector.getAmplifiers():
+            outputStats[amp.getName()] = {}
+
+        return outputStats
+
+    # Image measurement methods.
+    def imageStatistics(self, exposure, statControl):
```

Comment:
You lost the second "s" on `statsControl` from above (and I still think it might be easier to just call them all `statisticsControl`...but I only insist on consistency 😉 ).